vcd/resource_vcd_network_routed.go

```diff
@@ -214,6 +246,27 @@ func resourceVcdNetworkRoutedCreate(d *schema.ResourceData, meta interface{}) er
 		},
 		IsShared: d.Get("shared").(bool),
 	}
+	distributedAllowed := false
+	if edgeGateway.EdgeGateway.Configuration.DistributedRoutingEnabled != nil {
+		if *edgeGateway.EdgeGateway.Configuration.DistributedRoutingEnabled {
+			distributedAllowed = true
+		}
+	}
+	networkInterface := d.Get("interface_type").(string)
+	trueValue := true
+	switch strings.ToLower(networkInterface) {
+	case "internal":
+		// default: no configuration is needed
+		orgVDCNetwork.Configuration.SubInterface = nil
+	case "subinterface":
+		orgVDCNetwork.Configuration.SubInterface = &trueValue
+	case "distributed":
+		if distributedAllowed {
+			orgVDCNetwork.Configuration.DistributedInterface = &trueValue
+		} else {
+			return fmt.Errorf("interface 'distributed' requested, but edge gateway '%s' not enabled", edgeGateway.EdgeGateway.Name)
```

Comment:
I'd think this could be rephrased to better. Currently I read the error - "edge gateway is not enabled" while in fact "distributed routing is not enabled on edge gateway"

Reply:
Thanks. Changed